tests/java/lang/bad_hex_conversion/testdata/main.java

```diff
@@ -0,0 +1,41 @@
+package strings;
+
+import java.security.MessageDigest;
+
+public class BadHexConversion {
+
+    public String bad(String text) {
+        MessageDigest md = MessageDigest.getInstance("SHA-256");
+        byte[] resultBytes = md.digest(text.getBytes("UTF-8"));
+
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0, resultBytesLength = resultBytes.length; i < resultBytesLength; i++) {
+            byte b = resultBytes[i];
+            // bearer:expected java_lang_bad_hex_conversion
+            String badHex = Integer.toHexString(b);
+
+            // bearer:expected java_lang_bad_hex_conversion
+            stringBuilder.append(Integer.toHexString(resultBytes[i]));
+        }
+
+        for (byte b : resultBytes) {
+          // TODO: expected java_lang_bad_hex_conversion
+          stringBuilder.append(Integer.toHexString(b));
+        }
```

Comment:
Are we able to match on these yet? /cc @cfabianski or @didroe when you've a moment

Reply:
I'll work on a fix for it

Reply:
Yes. I just pushed the fix in the rule and in the test case.